### PR TITLE
Replicate dashboard chart animations and axis labelling on account transaction pages #634

### LINKED ISSUE #634: Replicate dashboard chart animations and axis labelling on account transaction pages
The dashboard time-series cards (`TimeSeriesValueCard`) have a polished chart treatment that the account transaction pages (`AccountDetailsHero`, used by the currency, bond and investment account details pages) do not share:

- **Draw animation** — dashboard charts animate in (`Animations { Enabled = true, Speed = 400 }`); the account hero chart has animations explicitly disabled, so the series snaps into place.
- **X axis labelling** — dashboard charts render datetime ticks with a per-granularity formatter (`dd MMM` / `MMM 'yy` / `yyyy`); the hero chart hides x-axis labels entirely, so there is no time reference under the curve.
- **Y axis labelling** — dashboard charts render compact floating currency ticks (`2.5k`, `10k`, …) overlaid on the plot with a padded, non-zero-based scale; the hero chart hides the y-axis, so the curve has no value reference.

Bring the account transaction page chart in line with the dashboard cards: enable the same draw animation and add matching X and Y axis labelling (same colours, sizes, formatters, floating y-axis, padded y-scale and gridlines).

### Acceptance criteria

- The account details hero chart animates on draw and on range change, matching the dashboard cards.
- X axis shows datetime tick labels formatted by granularity.
- Y axis shows compact floating currency tick labels with the same padded scale used by the dashboard cards.
- Labels stay legible over the gradient fill in both light and dark themes, on mobile and desktop.

### PR DESCRIPTION
**Pull request by avresial**
closes #634

Brings the balance chart on the account transaction pages (currency, bond and investment account details) in line with the dashboard time-series cards.

## What changed

**Draw animation** — the hero chart had animations explicitly disabled, so the series snapped into place. It now uses the same `Enabled = true, Speed = 400` animation as the dashboard cards, replaying on mount and on every range change.

**X axis labelling** — labels were hidden entirely. The axis is now a datetime axis with the dashboard's per-granularity formatter (`dd MMM` / `MMM 'yy` / `yyyy`), same 11px muted colour, borders and ticks off, crosshair on.

**Y axis labelling** — the axis was hidden. It is now a floating labelled axis overlaying the plot (so the full-bleed chart keeps its edge-to-edge width), with horizontal gridlines and the dashboard's padded, non-zero-based scale so small balance movements stay visible.

**Tick precision adapts to the account.** A literal copy of the dashboard's compact `k` formatter is wrong here: dashboard cards show portfolio-scale aggregates, but an account balance can be small. On the seeded 300 PLN bond account every tick collapsed to the same `0.3k`. `ChartHelper.GetYAxisTickFormatter` picks the formatter from the tick spacing — hundreds-scale gaps keep the compact `28k` / `-22k` form, tighter ranges fall back to plain numbers with just enough decimals (`300.17`, `300.38`).

**Legend moved to the top right** on the investment account, so the benchmark legend no longer competes with the new x-axis label row.

**Fixed a latent staleness bug.** The chart is repainted from a cached snapshot and then updated with the refreshed series *without* toggling `IsChartLoading` in between, so the `ApexChart` component is not remounted and that second update was silently dropped. The component now tracks the drawn points and calls `RenderAsync()` when they change — the same guard `TimeSeriesValueCard` already uses.

## Shared code

`AddYRangePadding` moved from `TimeSeriesValueCard` into `ChartHelper` alongside the compact tick formatter, so both charts scale and label from one place. The dashboard's rendered output is unchanged.

## Verification

- `dotnet build` clean (warnings-as-errors), `dotnet format` applied.
- Unit tests green: 725 passed. Tests for `AddYRangePadding` moved to `ChartHelperTests`, plus new coverage for the tick-formatter selection.
- Rendered on the guest account and screenshotted on mobile (430px) and desktop (1280px) across the cash, loan, bond and investment account pages, and the dashboard to confirm no regression there.
- Probed the live chart config in the browser to confirm the options actually reach ApexCharts: `animations {enabled: true, speed: 400}`, datetime x-axis with the formatter, floating labelled y-axis with the padded min/max.

---
_Generated by [Claude Code](https://claude.ai/code/session_01NezHftsUN9HRwTv1CNkbRA)_

## Summary by CodeRabbit

- **Enhancements**
  - Improved account detail charts with more responsive rendering when financial data changes.
  - Added clearer axis scaling, compact currency labels, date formatting, crosshairs, and refined chart visuals.
  - Improved chart readability over gradient backgrounds and enhanced pointer interactions.
  - Standardized chart formatting and spacing across time-series value cards.

- **Tests**
  - Added coverage for axis padding and adaptive currency tick formatting, including edge cases.

### CODE REVIEW

File: code/FinanceManager.Components/Shared/Helpers/ChartHelper.cs [hunk 1]
```diff
@@ -2,10 +2,48 @@ namespace FinanceManager.Components.Shared.Helpers;
...
 public static class ChartHelper
 {
+    /// <summary>
+    /// Compact currency tick labels for a y-axis: 2.5k / 7.5k / 10k / 13k. Shared by the
+    /// dashboard time-series cards and the account details hero chart so both label their
+    /// value axis identically.
+    /// </summary>
+    public const string CompactCurrencyTickFormatter =
+        "function(v){ if(!v) return ''; var k=v/1000; " +
+        "return (Math.abs(k)<10 && k%1!==0 ? k.toFixed(1) : Math.round(k))+'k'; }";
```
[the remark this answers is not in the document]
Comment (avresial) [reply code/FinanceManager.Components/Shared/Helpers/ChartHelper.cs:13]: Fixed in a6837dc — valid catch. `!v` is falsy for `0`, so an axis straddling zero lost that gridline's label. That mattered little on the dashboard aggregates this formatter came from, but account charts hit zero routinely (a loan paid off, an emptied cash account).

I used a slightly different guard than suggested:

```js
if(!Number.isFinite(v)) return ''; if(v===0) return '0';
```

`v===undefined || v===null` would have dropped the `NaN` case that `!v` was previously covering, and `NaN` falls through to `Math.round(NaN)+'k'` → `"NaNk"`. `Number.isFinite` covers `null`, `undefined` and `NaN` in one check.

Added two regression tests: one asserting zero is labelled rather than blanked, one asserting the non-numeric guard survives, so a future simplification back to `if(!v)` fails the build.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: code/FinanceManager.Components/Features/FinancialAccounts/Components/Shared/AccountDetailsHero.razor.cs [hunk 2]
```diff
@@ -38,36 +38,158 @@ public partial class AccountDetailsHero
...
+            // skeleton) has just drawn the current data itself.
+            _drawnChart = _chart;
+            _drawnPoints = [.. points];
+            return;
+        }
+
+        if (points.SequenceEqual(_drawnPoints)) return;
+
+        _drawnPoints = [.. points];
+        await _chart.RenderAsync();
```
[the remark this answers is not in the document]
Comment (avresial) [reply code/FinanceManager.Components/Features/FinancialAccounts/Components/Shared/AccountDetailsHero.razor.cs:94]: Fixed in a6837dc. The concrete hole: `ChartData = [A, B]` with an empty benchmark flattens to the same sequence as `ChartData = [A]` with `BenchmarkData = [B]`, so a moved series boundary compares equal and the redraw is skipped. A renamed benchmark with unchanged values has the same problem — the legend goes stale.

Both need an unlikely coincidence to hit in practice, but the guard is only correct by accident without this, and tagging each point costs nothing in a comparison that already materialises the list.

One deviation from the suggestion: I pulled `"Balance"` into a `BalanceSeriesName` constant and bound the `.razor` to it rather than hardcoding the literal in the tracking code. The name was already a literal in the markup, and adding a second copy is exactly the kind of pair that drifts.

Verified the legend still renders `Balance` / `Polish inflation` on the investment account after the binding change.

---
_Generated by [Claude Code](https://claude.ai/code)_
